### ISSUE TITLE
[Question] Microsoft Graph support with AAD authentication

### ISSUE THREAD

**Issue by hraz-msft** [Azure/static-web-apps#189]
First thing, Azure Static WebApps is absolutely AMAZING service!

For the question: 
Can we access to Microsoft Graph on-behalf-of the user if he was authenticated with the AAD provider? If so, how can we do that?

**Comment by anthonychu** [comment]
I don't believe this is possible right now because we don't have a way to expose the token to the function app.

**Comment by jeroenvdmeer** [comment]
@anthonychu any changes in the plans since your previous comment? I have the same use case as @hrazmsft. As users have already logged in to access my application, it's confusing for them to log in again to access MS Graph. Exposing the access token would be really great.

**Comment by JeremiahS-KMC** [comment]
I would also like to have this as well. My app connects to an API that requires user tokens to be set with requests. Right now I have to use a framework to handle Azure AD login so that my app can get access to the user token to pass with the request. Would be nice to just let the static apps service handle auth and get the user token from it in my app.

**Comment by diberry** [comment]
@Anthonychu - if it isn't possible - can you explain what Azure architectural elements would handle this right now?

**Comment by anthonychu** [comment]
We released a feature to programmatically assign roles with a function. Within the function, you have access to the token from the IdP. See the [documentation](https://docs.microsoft.com/azure/static-web-apps/authentication-authorization?tabs=function#role-management) for more details. There's also a [tutorial](https://docs.microsoft.com/azure/static-web-apps/assign-roles-microsoft-graph) with a sample app that shows you how to use the token to make a call to Microsoft Graph.

In the role assignment function, you could save the token for use later in another function. Would that work for you?

**Comment by hrazmsft** [comment]
Yes, sounds good

**Comment by ksdaniel** [comment]
Not sure if this discussion still is relevant - but the token has a 90 minutes life span.

The 'get roles' api endpoint gets called once - when the user logs in. 

This would mean - 90 minutes after the user logs in - the token expires and the calls to API's that rely on that token for authentication will start failing. 

Am I missing something? Has there been some improvements on this topic in the last 2 years? 



> We released a feature to programmatically assign roles with a function. Within the function, you have access to the token from the IdP. See the [documentation](https://docs.microsoft.com/azure/static-web-apps/authentication-authorization?tabs=function#role-management) for more details. There's also a [tutorial](https://docs.microsoft.com/azure/static-web-apps/assign-roles-microsoft-graph) with a sample app that shows you how to use the token to make a call to Microsoft Graph.
> 
> In the role assignment function, you could save the token for use later in another function. Would that work for you?
